File: code/app/src/main/java/com/example/avtokampi_android_app/models/Rezervacija.kt
package com.example.avtokampi_android_app.models

import java.time.LocalDateTime

data class Rezervacija(
    val id: Int,
    val datumOd: LocalDateTime,
    val datumDo: LocalDateTime,
    val createdAt: String,
    val updatedAt: String,
    val uporabnik: Uporabnik,
    val avtokamp: Avtokamp,
    val kampirnoMesto: KampirnoMesto,
    val vrstaKampiranja: VrstaKampiranja,
    val statusRezervacije: StatusRezervacije
)

File: avtokampi/lib/models/Rezervacija.dart
class Rezervacija {
    int id;
    String datumOd;
    String datumDo;
    String createdAt;
    String updatedAt;
    int uporabnik;
    int avtokamp;
    int kampirnoMesto;
    int vrstaKampiranja;
    int statusRezervacije;

    Rezervacija.nov();

    Rezervacija(this.id,
        this.datumOd,
        this.datumDo,
        this.createdAt,
        this.updatedAt,
        this.uporabnik,
        this.avtokamp,
        this.kampirnoMesto,
        this.vrstaKampiranja,
        this.statusRezervacije);

    Rezervacija.nova(this.datumOd,
        this.datumDo,
        this.uporabnik,
        this.avtokamp,
        this.kampirnoMesto,
        this.vrstaKampiranja,
        this.statusRezervacije);

    static Rezervacija fromJson(json) {
        Rezervacija r = new Rezervacija.nov();
        r.id = json['rezervacijaId'];
        r.datumOd = json['trajanjeOd'];
        r.datumDo = json['trajanjeDo'];
        r.createdAt = json['createdAt'];
        r.updatedAt = json['updatedAt'];
        r.uporabnik = json['uporabnik'];
        r.avtokamp = json['avtokamp'];
        r.kampirnoMesto = json['kampirnoMesto'];
        r.vrstaKampiranja = json['vrstaKampiranja'];
        r.statusRezervacije = json['statusRezervacije'];
        return r;
    }
}
